### PR TITLE
Enable F# 9.0

### LINKED ISSUE #17499: Struct UnionCase doesn't seem to be a valid target for the DefaultAugmentationAttribute
Repro:
This fails to build with F# 9.0 (or preview) enabled

[<Struct>]
[<DefaultAugmentation(false)>]
type Foo =
    | Baz of int
    | Bat
    | Batman

let foo = [Baz 42; Bat; Batman]
printf "%A" foo
c:\temp>c:\kevinransom\fsharp\artifacts\bin\fsc\Release\net472\fsc foo.fs
Microsoft (R) F# Compiler version 12.9.100.0 for F# 9.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

foo.fs(2,3): error FS0842: This attribute is not valid for use on this language element

### PR DESCRIPTION
**Pull request by KevinRansom**
Fixes https://github.com/dotnet/fsharp/issues/16739

### CODE REVIEW

**Comment by Martin521** [comment]
Will `supportsNonStringArguments` not be part of F# 9.0?

**Review by psfinaki — APPROVED** [review]
Don't know the context for all the changes, but where I know it - LGTM.

**Review by edgarfgp — APPROVED** [review]
From what I contributed it looks ok 👍🏻

**Review by vzarytovskii — APPROVED** [review]

File: docs/release-notes/.FSharp.Compiler.Service/9.0.100.md [hunk 1]
```diff
@@ -22,5 +22,5 @@
 * Optimize metadata reading for type members and custom attributes. ([PR #17364](https://github.com/dotnet/fsharp/pull/17364))
 * Enforce `AttributeTargets` on unions. ([PR #17389](https://github.com/dotnet/fsharp/pull/17389))
 * Ensure that isinteractive multi-emit backing fields are not public. ([Issue #17439](https://github.com/dotnet/fsharp/issues/17438)), ([PR #17439](https://github.com/dotnet/fsharp/pull/17439))
-
+* Enable FSharp 9.0 Language Version ([Issue #17497](https://github.com/dotnet/fsharp/issues/17438)), [PR](https://github.com/dotnet/fsharp/pull/17500)))
```
Comment (pedersorensen) [inline docs/release-notes/.FSharp.Compiler.Service/9.0.100.md:25]: Hi
Not sure if I'm approaching this issue correctly, but the link points to #17438 while the display text is #17497.

Comment (psfinaki) [reply docs/release-notes/.FSharp.Compiler.Service/9.0.100.md:25]: Thanks for noticing, fixing it [here](https://github.com/dotnet/fsharp/pull/17883).

**Pull request merged**